packages/shorebird_cli/lib/src/shorebird_flutter.dart

```diff
@@ -218,4 +218,31 @@ class ShorebirdFlutter {
     shorebirdEnv.flutterRevision = revision;
     useFlutterProgress.complete();
   }
+
+  /// Returns an human readable string from the received [flutterRevision] and
+  /// [flutterVersion].
+  ///
+  /// If [flutterRevision] is null, the current revision will be used.
+  /// If [flutterVersion] is null, it will be calculated from the current
+  /// revision using [getVersionString].
+  Future<String> humanReadableVersion({
+    String? flutterRevision,
+    String? flutterVersion,
+  }) async {
+    final revision = flutterRevision ?? shorebirdEnv.flutterRevision;
+    var version = 'unknown';
+
+    if (flutterVersion != null) {
+      version = flutterVersion;
+    } else {
+      try {
+        final calculatedValue = await getVersionString(revision: revision);
+        if (calculatedValue != null) {
+          version = calculatedValue;
+        }
+      } catch (_) {}
+    }
```

Comment:
```suggestion
    final String resolvedFlutterVersion;
    try {
      resolvedFlutterVersion = flutterVersion ?? await getVersionString(revision: revision);
    } catch (_) {
      resolvedFlutterVersion = 'unknown';
    }
```

Reply:
This will not work, `getVersionString` can `null`. 

The current choice in the source base was because I didn't wanted to repeat the string `unknown`

Reply:
New suggestion:

```suggestion
    final String resolvedFlutterVersion;
    try {
      resolvedFlutterVersion = flutterVersion ?? await getVersionString(revision: revision) ?? 'unknown';
    } catch (_) {
      resolvedFlutterVersion = 'unknown';
    }
```

Even if we have to repeat unknown, simplifying the control flow makes this a ton easier to read.

Reply:
We can also pull `unknown` into a constant 🤷

Reply:
I didn't changed to the suggested code, but I did change to something which I believe is better, let me know with you agree.

Reply:
The branching code looks to be the same as it was when I originally reviewed. I still prefer what I suggested.

Reply:
To elaborate a bit, the reasons I prefer the non-branching option:

1. There is less cognitive load on the reader due to `resolvedFlutterVersion` being final and only to assigned once.
2. It's shorter
3. It makes the fallback that's happening from flutterVersionArg -> getVersionString -> 'unknown' much more explicit.